core/database/arango_db_manager.py

```
@@ -0,0 +1,252 @@
+==============================================
+
+Shared database management module for all HADES-Lab processing pipelines.
+Provides connection pooling, transaction support, and distributed locking.
+Used by ArXiv, GitHub, and other data source processors.
+"""
+
+import os
+import time
+import random
+import logging
+from typing import Dict, Any
```

Ruff (0.12.2)

`typing.Dict` is deprecated, use `dict` instead

(UP035)